pil-analyzer/src/type_inference.rs

```diff
@@ -569,7 +575,15 @@ impl TypeChecker {
                     .map(|item| self.infer_type_of_expression(item))
                     .collect::<Result<_, _>>()?,
             }),
-            Expression::LambdaExpression(_, LambdaExpression { kind, params, body }) => {
+            Expression::LambdaExpression(
+                _,
+                LambdaExpression {
+                    kind,
+                    params,
+                    body,
+                    param_types,
```

Comment:
can this `param_types` already have some value here? it seems to be overridden without being looked at

Reply:
Will add an assertion.